### PR TITLE
pkp/pkp-lib#2102 Implement SelectAll feature for SelectListPanel.

### PR DESCRIPTION
**Pull request by NateWr**
- Adds SelectListPanelSelectAll component.
- Implements select all for submission list panels.
- Refactors ListPanel notices into ListPanelNotice component.

### CODE REVIEW

File: src/components/ListPanel/submissions/SubmissionsListPanel.vue [hunk 1]
```diff
@@ -23,6 +23,12 @@
 			/>
 		</div>
 		<div class="pkpListPanel__body -pkpClearfix pkpListPanel__body--submissions">
+			<select-list-panel-select-all
+				v-if="showSelectAll"
+				:label="i18n.selectAllLabel"
+				:checked="selectAllChecked"
+				@toggle="toggleSelectAll"
+			/>
```
Comment (bozana) [inline src/components/ListPanel/submissions/SubmissionsListPanel.vue]: why this select-all feature in the SubmissionsListPanel?

File: src/components/SelectListPanel/SelectListPanelItem.vue [hunk 2]
```diff
@@ -67,26 +62,23 @@ export default {
...
+	methods: {
 		/**
-		 * Set an item to selected when loaded
+		 * Emit an event to select or de-select this item
 		 */
-		if (this.item._initializeSelected) {
-			this.selected = this.inputType === 'radio' ? this.inputValue : true;
-		}
+		toggle: function () {
+			this.$emit('toggle', this.inputValue);
```
Comment (bozana) [inline src/components/SelectListPanel/SelectListPanelItem.vue]: where is then this selected/de-selected?

File: src/components/ListPanel/submissions/SubmissionsListPanel.vue (hunk 1 above)
Comment (NateWr) [reply src/components/ListPanel/submissions/SubmissionsListPanel.vue]: Ahh! You're right. I don't know what I was thinking. :)

https://github.com/NateWr/ui-library/commit/e8e13b4b83b2cd61b0009666764e2469592dbfa1

File: src/components/SelectListPanel/SelectListPanelItem.vue (hunk 2 above)
Comment (NateWr) [reply src/components/SelectListPanel/SelectListPanelItem.vue]: The parent component, `SelectListPanel`, listens for the event and toggles the value here:

https://github.com/NateWr/ui-library/blob/select_all/src/components/SelectListPanel/SelectListPanel.vue#L27

Vue.js follows a pattern where data needs to be changed at the highest-level component that controls it. Data that is "passed down" to a child component shouldn't be modified in that child component. When a child component wants to modify that data, it needs to emit an event to the parent, which updates the data.

https://vuejs.org/v2/guide/components.html#Composing-Components

**Pull request merged**